add docs for the Navigation API (#21960)
mdn/content: files/en-us/web/api/navigation/traverseto/index.md

---
title: Navigation.traverseTo()
slug: Web/API/Navigation/traverseTo
page-type: web-api-instance-method
tags:
  - API
  - Experimental
  - History
  - Method
  - Navigate
  - Navigation
  - Navigation API
  - Reference
  - Scroll
  - transition
  - Traversal
  - traverseTo
browser-compat: api.Navigation.traverseTo
---

{{APIRef("Navigation API")}}{{seecompattable}}

> The **`traverseTo()`** method of the
> {{domxref("Navigation")}} interface navigates to a specific {{domxref("NavigationHistoryEntry")}} by {{domxref("NavigationHistoryEntry.key", "key")}}.

Reviewer: ```suggestion
The **`traverseTo()`** method of the {{domxref("Navigation")}} interface navigates to the {{domxref("NavigationHistoryEntry")}} identified by the given {{domxref("NavigationHistoryEntry.key", "key")}}.
```

(I think?)

## Syntax

```js-nolint

> traverseTo(key, options)

Reviewer: ```suggestion
traverseTo(key)
traverseTo(key, options)
```

```

### Parameters

- `key`
  - : The `key` of the {{domxref("NavigationHistoryEntry")}} to navigate to.
- `options` {{optional_inline}}
  - : An options object containing the following properties:
    - `info`
      - : Developer-defined information to be passed along to the {{domxref("Navigation/navigate_event", "navigate")}} event, made available in {{domxref("NavigateEvent.info")}}. This can be any data type. You might, for example, wish to display newly-navigated content with a different animation depending on how it was navigated to (swipe left, swipe right, or go home). A string indicating which animation to use could be passed in as `info`.

### Return value

> An object containing two promises — `{ committed, finished }`. This allows the invoking function to wait on taking further action until:
> 
> - `committed` fulfills, meaning that the visible URL has changed and a new {{domxref("NavigationHistoryEntry")}} has been created.
> - `finished` fulfills, meaning that all promises returned by the `intercept()` handler are fulfilled. This is equivalent to the {{domxref("NavigationTransition.finished")}} promise fulfilling, when the {{domxref("Navigation/navigatesuccess_event", "navigatesuccess")}} event fires.
> - either one of the above promises rejects, meaning that the navigation has failed for some reason.

Reviewer: ```suggestion
An object with the following properties:

- `committed`
  - : A {{jsxref("Promise")}} which will fulfill when the visible URL has changed and a new {{domxref("NavigationHistoryEntry")}} has been created.
- `finished`
  - : A {{jsxref("Promise")}} which will fulfill when all promises returned by the `intercept()` handler are fulfilled. This is equivalent to the {{domxref("NavigationTransition.finished")}} promise fulfilling, when the {{domxref("Navigation/navigatesuccess_event", "navigatesuccess")}} event fires.
 
Either one of these promises rejects if the navigation has failed for some reason.
```
